### DISCUSSION TITLE
Section Editor Stack View

### DISCUSSION THREAD

**Discussion by rbarbour** [thinreports/thinreports#22] [Q&A] [answered]
Hello,
I have recently come across the new Section Editor, which solves a number of the limitations I hit with the original editor with the headers, footers, auto-stretch to better support my use case.
I have been trying to get up to speed with the editor and generator capability, and after adding localisation support to get over the initial language barrier (pull request raised), I am trying to build a similar report but have some questions around the Stack View.
In the previous editor, I used a List in my template, and add_row in my code to represent a table with my data, updating the text placeholders in the row. In the new editor, I am trying to do a similar thing with the Stack View, where the first row in the Stack View would be the header, with column headings, and then the next row is the template for the rows of data I want to generate automatically, with the row auto-stretching if necessary (in the previous editor I used Shrink to fit, but some data  edge cases cased the text to be too small to read). 
Is this type of capability currently supported (I have been crawling through the code, but thought it beneficial to ask), and if so, could you provide some information on how this might be achieved, as the documentation is quite light at present (happy to help with this once I get to grips with the functionality).
Thanks

**Comment by hidakatsuya** [comment, marked as answer]
Hello,
Thank you for your question. Also, thank you for your pull request 😄 

First of all, the StackView tool in the Section Report Editor and the List tool in the previous editor (called the Basic Report Editor) are similar, but their uses are different. And the List tool is not supported in the Section Report format.

The StackView tool is used to represent stacked layouts that are divided into units of rows. The rows can be dynamically hidden, but they cannot be dynamically added like the rows in the List tool. See the following sample for more details (sorry, it's in Japanese).
https://github.com/thinreports/thinreports-generator/tree/master/test/features/section_report_stack_view

In the Section Report format, you can create dynamic tables by combining header/footer/detail sections. For more details, please refer to the following sample code and sample PDF (sorry, it's in Japanese too).
https://github.com/thinreports/thinreports-generator/tree/master/test/features/section_report_basic

Does this answer your question?

**Reply by rbarbour** [reply]
Hello!
Thanks for coming back to me.
Apologies, I completely missed that a section can repeat, and hence assumed (incorrectly) that I should be further looking into the StackView, so this does indeed help me. 
I will give this a go, and let you know how I get on. 
Thanks 🥰

**Reply by rbarbour** [reply]
After some playing around, I was able to use headers and multiple details sections to output a table. 
Thanks again
